### ISSUE TITLE
Download folder can't be found -> nothing is imported, but download is removed from Activity page and can'd be manually imported.

### ISSUE THREAD

**Issue by dawtmaytrikx** [Lidarr/Lidarr#2746]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I use a seedbox but run lidarr on my NAS. Once the download completes a script moves the files from the seedbox to the NAS. However, as soon as the download completes, lidarr throws the following error:

`DownloadTrackImportService       Import failed, path does not exist or is not accessible by Lidarr: /downloads/torrent/complete/<...>. Ensure the path exists and the user running Lidarr has the correct permissions to access this file/folder`

Lidarr also changes the category of the torrent to the post-import category and removes the download from the Activity tab, which means I have to manually copy the files to the right location.

### Expected Behavior

I expect the download to remain in the Activity tab until files are found, and then automatically imported. As far as I can tell this should have been resolved with https://github.com/Lidarr/Lidarr/issues/945

### Environment

```markdown
- OS: Debian 11
- Lidarr: 1.0.0.2515
- Docker Install: Yes
- Using Reverse Proxy: Yes
- Browser: not relevant
```

### What branch are you running?

Nightly

### Trace Logs?

Sorry, lidarr is completely unresponsive when activating trace logs, so here are some debug logs instead.
[lidarr.debug.log](https://github.com/Lidarr/Lidarr/files/8457373/lidarr.debug.log)

**Comment by bakerboy448** [comment]
cc @starstruct
env from #4342
```
- OS: Ubuntu 22.04
- Lidarr: 1.4.5.3639
- Docker Install: No
- Using Reverse Proxy: No
- Browser: Firefox
- Database: sqlite
```

logs from #4342

```
lidarr.trace.28.txt:2023-11-29 22:35:23.8|Info|DownloadService|Report sent to rtorrent from indexer private_index Prowlarr. Chevelle - Vena Sera [2007] [Album] [FLAC Lossless] [CD] [Cue]
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Trace|RemotePathMappingService|Evaluating remote path remote mappings for match to host [my.server.com] and remote path [/home/(removed)/torrents/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]]
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|RemotePathMappingService|Remapped remote path [/home/(removed)/torrents/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]] to local path [/mnt/autofs/remote/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]] for host [my.server.com]
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Parsing string 'Chevelle - Vena Sera [2007] [Album] [FLAC Lossless] [CD] [Cue]'
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Album Parsed. Chevelle - Vena Sera 
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|QualityParser|Trying to parse quality for 'Chevelle - Vena Sera [2007] [Album] [FLAC Lossless] [CD] [Cue]'
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Parsing string 'Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]'
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Album Parsed. Chevelle - Vena Sera 
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|QualityParser|Trying to parse quality for 'Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]'
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|TrackedDownloadService|Tracking '	rtorrent:Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]': ClientState=Completed (readonly) LidarrStage=Downloading Album='Chevelle - Vena Sera FLAC v1' OutputPath=/mnt/autofs/remote/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac].
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Parsing string 'Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]'
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|Parser|Album Parsed. Chevelle - Vena Sera 
lidarr.trace.27.txt:2023-11-29 22:36:24.6|Debug|QualityParser|Trying to parse quality for 'Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]'
lidarr.trace.27.txt:2023-11-29 22:36:24.7|Debug|DownloadedTracksImportService|Processing path: /mnt/autofs/remote/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]
lidarr.trace.27.txt:2023-11-29 22:36:24.7|Error|DownloadedTracksImportService|Import failed, path does not exist or is not accessible by Lidarr: /mnt/autofs/remote/rtorrent/Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]. Ensure the path exists and the user running Lidarr has the correct permissions to access this file/folder
lidarr.trace.27.txt:2023-11-29 22:36:24.7|Debug|CompletedDownloadService|All albums were imported for Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]
lidarr.trace.27.txt:2023-11-29 22:36:24.7|Debug|DownloadEventHub|[Chevelle -Vena Sera (2007){Epic 88697 02698 2 CD}[Flac]] Marking download as imported from rtorrent
```

**Comment by starstruct** [comment]
More details for https://github.com/Lidarr/Lidarr/issues/4342:

I've been experimenting with timeouts for my mounts. 

I use `autofs` to control the mounting, and it does the mounting with `rclone`. 

Rclone caches the contents of the remote directory to make searching faster.  So I set rclone to refresh the directory every 30 seconds (dir-cache-timeout=0m30s). However, I don't want to constantly refresh the remote directory for perpetuity, so I set autofs to disconnect the mount after 300 seconds. This keeps the directory contents slightly more up-to-date while the drive is mounted.

This has improved my imports by about 50%. If I set `dir-cache-timeout` lower, I think it will solve my particular issue with imports. 

**However, the original issue still exists. Lidarr should mark failed imports as failures (rather than reporting them as successfully imported).**

**Comment by GregTroar** [comment]
Hi all,

I have same issue, local Lidarr, remote seedbox.
Is there any plan to solve this ?

Thanks

**Comment by hgfi** [comment]
Hi,

Exact same issue here. My setup:

NAS with lidarr installed under docker.
cron script that rsync every 5 minutes between my seedbox and my NAS

When a download is completed on the seedbox, lidarr tries to find in and fails because it's not rsynced yet, so it logs an error that it can't access the folder yet it marks the album as imported successfully and doesn't try to import it ever again.

Radarr and sonarr handles this properly, the import job stays active with the message "Downloaded  - Waiting to import" and once it's rsynced it works just fine.

I checked the VerifyImport function inside CompletedDownloadService.cs for both radarr and lidarr and the logic for checking if the media imported successfully is different. My guess is lidarr does it wrong.

**Comment by Techman** [comment]
I am also facing the same issue. I use Syncthing to sync remote directories instead of rsync. As others have stated, Lidarr will write an error to its log about the folder missing but still mark the download as successfully imported.

**Comment by bakerboy448** [comment]
This is a known issue. Additional comments to the effect of me too are spam and noise. L

**Comment by starstruct** [comment]
There are two things to consider based on these logs:

1) Lidarr thinks the file doesn't exist (probably related to caching or latency from remote mounts. Will look into that later)
2) Lidarr logs an error for inaccessible path but also marks the import as successful -- **this is the real problem**


```
DownloadedTracksImportService says processing the path..
DownloadedTracksImportService says the path is inaccessible 
CompletedDownloadService says all albums were imported
DownloadEventHub marks download as imported
```

The issue happens somewhere between `DownloadedTracksImportService` and `CompletedDownloadService`


This is where the inaccessible error happens:
https://github.com/Lidarr/Lidarr/blob/d2bd9b48492ee240253f31e4acf6fc4c966d31ec/src/NzbDrone.Core/MediaFiles/DownloadedTracksImportService.cs#L108-L111


`VerifyImport` could be an issue (I'm not a C# person, hard to tell what's happening here) : https://github.com/Lidarr/Lidarr/blob/d2bd9b48492ee240253f31e4acf6fc4c966d31ec/src/NzbDrone.Core/Download/CompletedDownloadService.cs#L167-L180

**Comment by bakerboy448** [comment]
> Lidarr logs an error for inaccessible path but also marks the import as successful -- this is the real problem

yes known bug and i believe what https://github.com/Lidarr/Lidarr/pull/4815 is trying to fix

**Comment by schmitzkr** [comment]
Hi there, is there a workaround for this issue while a fix is worked on?

**Comment by hgfi** [comment]
Hi, I use an sshfs mount. It's horribly slow but it works for lidarr.
